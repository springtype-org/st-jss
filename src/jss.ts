import { FontFaceProperties, KeyFrameProperties } from 'springtype-types';
import { API } from './interface/API';
import { ClassProperty } from './interface/ClassProperties';
import { makeStyles } from './makeStyles';
import { makeFont } from './makeFont';
import { makeAnimation } from './makeAnimation';

export const jss: API = {
  defaultUnit: 'px',
  classNamePrefix: 'st',
  instanceCounter: 0,
  makeAnimation: (animationName: string, style: Partial<KeyFrameProperties>) =>
    makeAnimation(animationName, style, jss),
  makeFont: (style: FontFaceProperties) => makeFont(style),
  makeStyles: <T extends string = string>(classProperty: ClassProperty<T>) => makeStyles<T>(classProperty, jss),
};
