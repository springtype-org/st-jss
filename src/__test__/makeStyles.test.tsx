import { jss } from '../jss';

describe('makeStyles', () => {
  it('check style output', () => {
    const [classes, style] = jss.makeStyles({
      container: {
        // Normal jss
        fontFamily: 'Nonito',
        '@media (min-device-width: 800px)': {
          paddingTop: 50,
          flex: 1,
          ':hover': {
            backgroundColor: 'red',
          },
          ' span': {
            color: 'black',
            height: 4,
            fontSize: 20,
          },
          '.libraryClass': {
            color: 'black',
            height: 4,
            fontSize: 20,
          },
        },
      },
      animation: {
        paddingTop: 50,
        animationIterationCount: 'infinite',
        animationDuration: '4s',
      },
    });
    expect(classes).toMatchSnapshot();
    const { children: childrenOfStyle, ...otherPropsOfStyle } = style;
    expect(childrenOfStyle[0]).toMatchSnapshot();
    expect(otherPropsOfStyle).toMatchSnapshot();
  });
});
