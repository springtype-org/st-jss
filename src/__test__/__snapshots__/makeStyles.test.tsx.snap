// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`makeStyles check style output 1`] = `
Object {
  "animation": "st-animation-2",
  "container": "st-container-1",
}
`;

exports[`makeStyles check style output 2`] = `
".st-container-1 {
  font-family: Nonito;
}
@media (min-device-width: 800px) {
  .st-container-1 {
    padding-top: 50px;
    flex: 1;
  }
  .st-container-1:hover {
    background-color: red;
  }
  .st-container-1 span {
    color: black;
    height: 4px;
    font-size: 20;
  }
  .st-container-1.libraryClass {
    color: black;
    height: 4px;
    font-size: 20;
  }
}
.st-animation-2 {
  padding-top: 50px;
  animation-iteration-count: infinite;
  animation-duration: 4s;
}"
`;

exports[`makeStyles check style output 3`] = `
Object {
  "attributes": Object {
    "type": "text/css",
  },
  "type": "style",
}
`;
